deepinterpolation/generator_collection.py

```diff
@@ -897,112 +919,147 @@ class OphysGenerator(SequentialGenerator):
     def __init__(self, json_path):
         "Initialization"
         super().__init__(json_path)
+        self._gpu_available = tf.test.is_gpu_available()
+        self._movie_data = None
+        if self._gpu_available:
+            self._batch_tensor_index = None # Current index cached as a batch tensor on the GPU
+            self._batch_tensor = None
+            self._batch_indices = None
+            self._input_indices = None
 
         # For backward compatibility
-        if "train_path" in self.json_data.keys():
-            self.raw_data_file = self.json_data["train_path"]
+        self.cache_data = self.json_data.get("cache_data", False)
+        self.raw_data_file = self.json_data.get(
+            "train_path", self.json_data.get("movie_path"))
+        if self._gpu_available:
+            self.gpu_cache_full = self.json_data.get("gpu_cache_full", False)
         else:
-            self.raw_data_file = self.json_data["movie_path"]
-
+            self.gpu_cache_full = False
+        self.normalize_cache = self.json_data.get("normalize_cache", False)
         self.batch_size = self.json_data["batch_size"]
-        movie_obj_point = h5py.File(self.raw_data_file, "r")
-        raw_data = movie_obj_point["data"]
+        self.movie_statistics_sample_size = \
+            self.json_data.get("movie_statistics_sample_size", 1000)
 
-        self.total_frame_per_movie = int(raw_data.shape[0])
+        self.total_frame_per_movie = int(self.movie_data.shape[0])
 
         self._update_end_frame(self.total_frame_per_movie)
         self._calculate_list_samples(self.total_frame_per_movie)
 
-        average_nb_samples = np.min([int(raw_data.shape[0]), 100])
 
-        # For backward compatibility
-        if "cache_data" in self.json_data.keys():
-            self.cache_data = self.json_data["cache_data"]
-        else:
-            self.cache_data = False
-
-        if self.cache_data:
```

Comment:
This section is important. It allows to cache only a part of the movie if the entire movie is not asked for.